sdk/src/metrics/data/circular_buffer.cc

```diff
@@ -0,0 +1,182 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/metrics/data/circular_buffer.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace metrics
+{
+
+namespace
+{
+
+struct AdaptingIntegerArrayIncrement
+{
+  size_t index;
+  uint64_t count;
+
+  template <typename T>
+  uint64_t operator()(std::vector<T> &backing)
+  {
+    const uint64_t result = backing[index] + count;
+    OPENTELEMETRY_LIKELY_IF(result <= uint64_t(std::numeric_limits<T>::max()))
+    {
+      backing[index] = static_cast<T>(result);
+      return 0;
+    }
+    return result;
+  }
+};
+
+struct AdaptingIntegerArrayGet
+{
+  size_t index;
+
+  template <typename T>
+  uint64_t operator()(const std::vector<T> &backing)
+  {
+    return backing[index];
+  }
+};
+
+struct AdaptingIntegerArraySize
+{
+  template <typename T>
+  uint64_t operator()(const std::vector<T> &backing)
+  {
+    return backing.size();
+  }
+};
+
+struct AdaptingIntegerArrayClear
+{
+  template <typename T>
+  void operator()(std::vector<T> &backing)
+  {
+    std::fill(backing.begin(), backing.end(), 0);
```

Comment:
Not sure, but you may want to typecast to `static_cast<T>(0)` to get rid of CI failure ?